### PR TITLE
upgrade go client to latest api spec using openapi-generator 5.x

### PR DESCRIPTION
**Pull request by houqp**
Depends on https://github.com/apache/airflow/pull/16411.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,19 +36,18 @@ import (
...
+	cli := airflow.NewAPIClient(conf)
 
 	cred := airflow.BasicAuth{
 		UserName: "username",
 		Password: "password",
 	}
 	ctx := context.WithValue(context.Background(), airflow.ContextBasicAuth, cred)
 
-	variable, _, err := cli.VariableApi.GetVariable(ctx, "foo")
+	variable, _, err := cli.VariableApi.GetVariable(ctx, "foo").Execute()
```
Comment (houqp) [inline README.md:50]: unfortunately openapi-generator 5.x introduced breaking change in their go code gen.

**Comment by msumit** [comment]
@houqp did u generate this code from Airflow's 2.1.0 tag or from master?

**Comment by houqp** [comment]
from master, I will regen it with `2.1.0`.

**Comment by potiuk** [comment]
Or `main` :)

**Comment by houqp** [comment]
my bad, takes time to retrain the muscle memory :P

**Comment by houqp** [comment]
alright, regenerated with 2.1.0 tag, ready for another round of review.

**Review by msumit — APPROVED** [review]

**Pull request merged**